plugins/rtmp-services/rtmp-common.c

```diff
@@ -446,6 +447,55 @@ static inline bool fill_twitch_servers(obs_property_t *servers_prop)
 	return success;
 }
 
+static bool fill_amazon_ivs_servers_locked(obs_property_t *servers_prop)
+{
+	struct dstr name_buffer = {0};
+	size_t count = amazon_ivs_ingest_count();
+	bool rtmps_available = obs_is_output_protocol_registered("RTMPS");
+
+	if (rtmps_available) {
+		obs_property_list_add_string(
+			servers_prop, obs_module_text("Server.AutoRTMPS"),
+			"auto-rtmps");
+	}
+	obs_property_list_add_string(
+		servers_prop, obs_module_text("Server.AutoRTMP"), "auto-rtmp");
+
+	if (count <= 1)
+		return false;
+
+	if (rtmps_available) {
+		for (size_t i = 0; i < count; i++) {
+			struct twitch_ingest ing = amazon_ivs_ingest(i);
```

Comment:
This adds an implicit dependency of the IVS ingest on the internal implementation of the `twitch_ingest` struct. Thus if one is changed, it will automatically break/change the Amazon IVS implementation.

While one might argue "they are effectively the same thing", that would lead to the question "then why is it exposed as two different implementations?".

IMO a copy of the struct should be declared for Amazon IVS to decouple the internal implementation details (which would be necessary once we move to true "service plugins" as each plugin needs to be an independent unit code-wise).

Reply:
As far as I can tell, this has been addressed.